### ISSUE TITLE
bug with : in regex attribute

### ISSUE THREAD

**Issue by izzabelle** [maciejhirsz/logos#68]
hi!

Writing a Lexer for a project and noticed a weird bug with the colon throwing an error when used in the regex attribute. not sure why this is happening and figured I should submit it here.

code:
```rs
enum Token {
    // ...
    #[regex = ".+:"]
    Label,
    // ...
}
```
rustc readout output:
```
error: proc-macro derive panicked
 --> src\assembler\lexer.rs:3:10
  |
3 | #[derive(Logos, PartialEq)]
  |          ^^^^^
  |
  = help: message: Internal Error: Invalid fork construction: [
              ':' -> TOKEN "Label",
              '[01-090B-FF]' -> REPEAT ['[01-090B-FF]'] -> ':' -> TOKEN "Label"
          ]

error: aborting due to previous error
```

**Comment by maciejhirsz** [comment]
Hey @nrlsk, does this happen on v0.10.0-rc1 or older? Can you paste the entire enum with all definitions (or provide a link to it)?

**Comment by nrlsk** [comment]
here's the entirety of the enum: https://gist.github.com/nrlsk/deed25c81628ab68f4826331d121811f

This is on the 0.9.7 build of logos.

**Comment by maciejhirsz** [comment]
Ok, I just tried running this on 0.10 and I get an infinite loop instead (also during compile time), so something fishy is happening here.

**Comment by nrlsk** [comment]
I tested it replacing `\w` with `.+` and it get's through compilation but still has that previously posted error. so it might be multiple things going on here.

**Comment by maciejhirsz** [comment]
This should be finally fixed in 0.10.0-rc3, please re-open if there are other issues.
